### PR TITLE
Pass aws_region_name to get_aws_service_client() in SageMakerLLM

### PR DESCRIPTION
**Pull request by a-a-ronchen**
# Description

2-line fix for #11999, allows the SageMakerLLM [example](https://docs.llamaindex.ai/en/stable/examples/llm/sagemaker_endpoint_llm.html)in the documentation to run without error.

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Review by hatianzhang — APPROVED** [review]

File: llama-index-integrations/llms/llama-index-llms-sagemaker-endpoint/llama_index/llms/sagemaker_endpoint/base.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def __init__(
         aws_access_key_id: Optional[str] = None,
         aws_secret_access_key: Optional[str] = None,
         aws_session_token: Optional[str] = None,
-        region_name: Optional[str] = None,
+        aws_region_name: Optional[str] = None,
```
Comment (logan-markewich) [inline llama-index-integrations/llms/llama-index-llms-sagemaker-endpoint/llama_index/llms/sagemaker_endpoint/base.py:86]: a tad weary of changing a core interface kwarg name. We might have to keep both and mark the old name as deprecated?

Comment (a-a-ronchen) [reply llama-index-integrations/llms/llama-index-llms-sagemaker-endpoint/llama_index/llms/sagemaker_endpoint/base.py:86]: @logan-markewich my rationale was to preserve consistency with the [documentation](https://docs.llamaindex.ai/en/stable/examples/llm/sagemaker_endpoint_llm.html). Anyone using sagemaker with llamaindex would have had to look at the source code and manually pass `region_name` as an kwarg (which is what we did and I was the first to raise the issue (#11999) alongside the PR). In my view adding extra code to pass `aws_region_name` by default and fall back to `region_name` if `aws_region_name` is `None` would also impact readability, but can do so if that's the direction you prefer.

Comment (logan-markewich) [reply llama-index-integrations/llms/llama-index-llms-sagemaker-endpoint/llama_index/llms/sagemaker_endpoint/base.py:86]: This is a breaking change to a very public interface -- I would prefer to give users some period of time to update their code. If they pass in the old kwarg, we should raise a warning that its deprecated

Comment (hatianzhang) [reply llama-index-integrations/llms/llama-index-llms-sagemaker-endpoint/llama_index/llms/sagemaker_endpoint/base.py:86]: another option is to pass this as a `kwargs` and make the old parameter as deprecated

**Review by logan-markewich — APPROVED** [review]

**Comment by brycecf** [comment]
@a-a-ronchen @logan-markewich Any updates on this?  Looks like a simple linting fix is all that is preventing this fix.

**Pull request merged**